addons/mail/models/res_partner.py

```diff
@@ -112,7 +112,13 @@ def get_mention_suggestions(self, search, limit=8):
         if len(users) < limit:
             partners = self.search_read(search_dom, fields, limit=limit)
             # Remove duplicates
-            partners = [p for p in partners if not len([u for u in users if u['id'] == p['id']])] 
+            partners = [p for p in partners if not len([u for u in users if u['id'] == p['id']])]
+
+        #add odoobot in mention suggestion when pinging in mail_thread
+        if len(partners) + len(users) < limit and "odoobot".startswith(search.lower()):
+            odoobot = self.env.ref("base.partner_root")
+            if not any(elem['id'] == odoobot.id for elem in partners):
+                partners.append({'id': odoobot.id, 'name': odoobot.name, 'email': odoobot.email})
```

Comment:
should use `odoobot.read(fields)[0]` to be more consistent with the rest of the method.